src/raftstore/store/peer.rs

```diff
@@ -703,8 +707,9 @@ impl Peer {
     }
 
     /// Collects all down peers.
-    pub fn collect_down_peers(&self, max_duration: Duration) -> Vec<PeerStats> {
+    pub fn collect_down_peers(&mut self, max_duration: Duration) -> Vec<PeerStats> {
         let mut down_peers = Vec::new();
+        let mut down_peer_ids = Vec::new();
```

Comment:
What about using `self.down_peer_ids`?

Reply:
Lifetime problems. Can't mutate it after L713.